community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -517,23 +517,27 @@ private void commit() throws TransactionFailureException
         try ( CommitEvent commitEvent = transactionEvent.beginCommitEvent() )
         {
             // Trigger transaction "before" hooks.
-            if ( hasDataChanges() )
+            if ( hasTxStateWithChanges() )
             {
-                try
+                if ( txState.hasDataChanges() )
                 {
```

Comment:
No, don't think so. `hasDataChanges` is there because it signals that there have been changes that are actually visible in a transaction event handler.